packages/core/test/multistep-dialog/multistepDialogTests.tsx

```diff
@@ -15,18 +15,47 @@
  */
 
 import { assert } from "chai";
-import { mount } from "enzyme";
+import { mount, ReactWrapper } from "enzyme";
 import * as React from "react";
 
 import { Classes, MultistepDialog, DialogStep } from "../../src";
+import { MultistepDialogProps } from "../../src/components/dialog/multistepDialog";
 
 const NEXT_BUTTON = "[text='Next']";
 const BACK_BUTTON = "[text='Back']";
 const SUBMIT_BUTTON = "[text='Submit']";
 
 describe("<MultistepDialog>", () => {
+    let wrapper: ReactWrapper<MultistepDialogProps, any, MultistepDialog>;
+    let isMounted = false;
+    const testsContainerElement = document.createElement("div");
+    document.documentElement.appendChild(testsContainerElement);
+
+    /**
+     * Mount the `content` into `testsContainerElement` and assign to local `wrapper` variable.
+     * Use this method in this suite instead of Enzyme's `mount` method.
+     */
+    function mountWrapper(content: JSX.Element) {
+        wrapper = mount(content, { attachTo: testsContainerElement });
+        isMounted = true;
+        return wrapper;
+    }
```

Comment:
these tests were causing Overlay tests to fail. see https://app.circleci.com/pipelines/github/palantir/blueprint/1539/workflows/f959324a-857b-4131-a89e-f3c315416103/jobs/46941: 

> \<Overlay\> Focus management does not focus overlay when closed - <Overlay> Focus management
> `Error: Uncaught AssertionError: expected <button></button> to equal <div class="bp3-dialog-container bp3-overlay-content bp3-overlay-appear-done bp3-overlay-enter-done" tabindex="0"><div class="bp3-dialog" role="dialog" style="min-width: 800px; padding-bottom: 0px;"><div class="bp3-multistep-dialog-panels"><div class="bp3-multistep-dialog-left-panel"></div></div></div></div>`

Note the **`bp3-multistep-dialog-left-panel`** class. The error suggests that something is causing these tests to affect the overlay tests but I'm not sure why...

Making this suite use a mountWrapper like the overlayTests seems to fix the issue.

Reply:
on closer inspection, I suspect the reason is due to some of these tests previously mounting but _not_ unmounting. For example: https://github.com/palantir/blueprint/blob/88a158b38a96350109650a7fe821a5091a25b3ed/packages/core/test/multistep-dialog/multistepDialogTests.tsx#L169-L180

Since this suite is alphabetically before overlayTests and because we've now fixed the `enforceFocus` behavior (which is enabled for both Dialog and MultistepDialog), these unmounted overlays are keeping focus. I'll revert this change and just unmount properly.

Reply:
unmounting properly sounds like a better change to do in this PR :+1: